src/components/data/_tests/data-utils.test.js

```diff
@@ -61,3 +61,79 @@ describe('concatenateAttributeNames', () => {
     expect(concatenateAttributeNames(attrList1, attrList2)).toEqual(expected)
   })
 })
+
+describe('getAttributeType', () => {
+  it('returns type of attribute value', () => {
+    expect(getAttributeType('value')).toEqual({ type: 'string', isList: false })
+    expect(getAttributeType(3)).toEqual({ type: 'number', isList: false })
+    expect(getAttributeType(false)).toEqual({ type: 'boolean', isList: false })
+    expect(getAttributeType({
+      entityType: 'thing',
+      entityName: 'thing_one'
+    })).toEqual({ type: 'reference', isList: false })
+
+    expect(getAttributeType({ items: ['a', 'b', 'c'] })).toEqual({ type: 'string', isList: true })
+    expect(getAttributeType({ items: [1, 2, 3] })).toEqual({ type: 'number', isList: true })
```

Comment:
What does it do with mixed types (`['a', 2, false]`, for example)?

Reply:
I _think_ that a list is not allowed to contain items of mixed types.

However, I'm not sure if it's possible to create a generic JSON value that looks like a list. Generic JSON values currently break this component ([AJ-342](https://broadworkbench.atlassian.net/browse/AJ-342)).

Reply:
Rawls does reject creating a list with mixed types. `batchUpsert` with:
```
[
  {
    "name": "thing_one",
    "entityType": "things",
    "operations": [
      {
        "op": "CreateAttributeValueList",
        "attributeName": "new_attr"
      },
      {
        "op": "AddListMember",
        "attributeListName": "new_attr",
        "newMember": "a"
      },
      {
        "op": "AddListMember",
        "attributeListName": "new_attr",
        "newMember": 2
      },
      {
        "op": "AddListMember",
        "attributeListName": "new_attr",
        "newMember": false
      }
    ]
  }
]
```
returns an error:
> inconsistent attributes for list: attribute lists must consist of a single data type. For attribute 'new_attr', found types: [AttributeString, AttributeNumber, AttributeBoolean]. Sample values for these types: [AttributeString(a), AttributeNumber(2), AttributeBoolean(false)]

There is still the issue of creating a raw JSON attribute that looks like a list, but I think maybe we can defer that to AJ-342 since this is no worse than the current behavior (https://github.com/DataBiosphere/terra-ui/blob/9053e0a383676f8ac183d278204c3e2ae35066be/src/components/data/data-utils.js#L478-L485).